lisa/tools/blobdownload.py

```diff
@@ -0,0 +1,115 @@
+import re
+from pathlib import PurePath
+from typing import Any, List
+
+from lisa.executable import Tool
+from lisa.util import LisaException
+
+from .python import Pip, Python
+
+
+class BlobDownloader(Tool):
+    def _initialize(self, *args: Any, **kwargs: Any) -> None:
+        self.pip = self.node.tools[Pip]
+        self.python = self.node.tools[Python]
+
+    @property
+    def command(self) -> str:
+        return "echo"
+
+    @property
+    def can_install(self) -> bool:
+        return True
+
+    def _check_exists(self) -> bool:
+        return (
+            self.python.run(
+                '-c "from azure.storage.blob import BlobServiceClient"',
+                force_run=True,
+            ).exit_code
+            == 0
+        )
+
+    def install(self) -> bool:
+        self.pip.install_packages("azure-storage-blob")
+        return self._check_exists()
+
+    def download_blob(
+        self,
+        connection_string: str,
+        container_name: str,
+        blob_name: str,
+        file_name: str = "",
+    ) -> PurePath:
+        if file_name:
```

Comment:
When is the file name or blob name provided? Can we just provide one way?

Reply:
update